### PR TITLE
clear activity task when restarting with an intent.

### LINKED ISSUE #22: Issue with activity stack when restarting with an intent.
Consider this change to the sample app:

Define `LaunhcerActivity` which is the launcher activity and it opens `MainActivity` inside of its `onCreate()`.
Change `restartWithIntent()` so that it creates intent with `LaunhcerActivity` class.

Now after clicking on `RESTART PROCESS WITH INTENT` button, it opens `LaunhcerActivity` then `LaunhcerActivity` opens `MainActivity`. Now if you press back you will see `LaunhcerActivity`, which is expected, but if you press back again, you will see another `MainActivity`.

It seems this is happening because `triggerRebirth(Context context, Intent... intents)` does not set `FLAG_ACTIVITY_NEW_TASK` and `FLAG_ACTIVITY_CLEAR_TASK` flags.

### PR DESCRIPTION
**Pull request by sadegh**
It resolves #22

### CODE REVIEW

File: process-phoenix/src/main/java/com/jakewharton/processphoenix/ProcessPhoenix.java [hunk 1]
```diff
@@ -53,10 +53,16 @@ public static void triggerRebirth(Context context) {
    * <p>
    * Behavior of the current process after invoking this method is undefined.
    */
-  public static void triggerRebirth(Context context, Intent... nextIntents) {
+  public static void triggerRebirth(Context context, Intent... intents) {
+    if (intents.length < 1) {
```
Comment (mradzinski) [inline process-phoenix/src/main/java/com/jakewharton/processphoenix/ProcessPhoenix.java]: By adding this line you are breaking backwards compatibility. As per the Readme, `ProcessPhoenix.triggerRebirth(context);` is a perfectly valid way of triggering the "rebirth".

Comment (sadegh) [inline process-phoenix/src/main/java/com/jakewharton/processphoenix/ProcessPhoenix.java]: @mradzinski that's why `public static void triggerRebirth(Context context){...}` exists as well. (line 47)

**Comment by JakeWharton** [comment]
I'm not opposed to this change, but I would like to replicate it with the sample first.

**Comment by JakeWharton** [comment]
Nevermind let's just take it

**Pull request merged**